route/build/src/pio_utils.f90

```diff
@@ -354,16 +366,16 @@ SUBROUTINE closeFile(pioFileDesc, fileOpen)
   END SUBROUTINE closeFile
 
   !-----------------------------------------------------------------------
-  SUBROUTINE freeDecomp(pioFileDesc, iodesc)
+  SUBROUTINE freeDecomp(pioIOsystem, iodesc)
```

Comment:
The thing is that using pioFileDesc wasn't working with the nag compiler. So I'd say we should only use pioIOsystem. It's possible there's a bug in using pioFileDesc -- just for this version of PIO though. But, since this is working with pioSystem, I'd say just use it.

Reply:
Ok, if it working as expected, i would say ok.  So CLM also uses pioSystem??

Reply:
I checked to see what CTSM is doing. It looks like it creates an array of io_desc_t types for the different dimensions and such it needs. But, it never free's the memory for them. So it's creating them once at initialization and not bothering to free them. I added issue #409 to talk about doing that in mizuRoute as well.

Reply:
I see.  so this can avoid memory leak from create/destroy repeat for each history file.   so one time initialization of pio decomposition would be way to go.